Library/Homebrew/os/mac/sdk.rb

```diff
@@ -52,7 +52,17 @@ def sdk_if_applicable(v = nil)
         rescue NoSDKError
           latest_sdk
         end
-        # Only return an SDK older than the OS version if it was specifically requested
+        # Accept an SDK for another OS version if it shares a major version
+        # with the current OS - for example, the 11.0 SDK on 11.1,
+        # or vice versa.
+        # Note that this only applies on macOS 11
+        # or greater, given the way the versioning has changed.
+        # This shortcuts the below check, since we *do* accept an older version
+        # on macOS 11 or greater if the major version matches.
+        return sdk if OS::Mac.version.major >= 11 && sdk.version.major == OS::Mac.version.major
+
+        # On OSs lower than 11, or where the major versions don't match,
+        # only return an SDK older than the OS version if it was specifically requested
         return unless v || (!sdk.nil? && sdk.version >= OS::Mac.version)
```

Comment:
My understanding from this is it already allows mismatched SDK versions. Can you explain what this covers and does not that the new code does?

Relatedly: could this be refactored to not use unless && or || (either use if or two statements).  Not your code I know but I find it really hard to parse this as-is.

Reply:
This allows picking an SDK which is *newer* than the one for the active OS, while explicitly disallowing SDKs *older* than the active OS unless explicitly requested. This handles the case where newer Xcode versions don't ship an SDK for the current OS and instead ship one for the latest OS, which we've seen often. For example, it's not uncommon for users on macOS 10.14 to only have an SDK for macOS 10.15, and so forth.

The case that the new case covers is cases where you have an *older* SDK than your current OS, but the only difference is the minor version and not the OS version. That's a case that can't occur before Big Sur.

Reply:
```suggestion
        return if v.blank? && sdk.present? && sdk.version < OS::Mac.version
```